cdap-docs/admin-manual/source/security/impersonation.rst

```diff
@@ -4,10 +4,42 @@
 
 .. _admin-impersonation:
 
-=========================
-Configuring Impersonation
-=========================
+=============
+Impersonation
+=============
 
-Impersonation allows users to run programs and access datasets, streams, and other resources as pre-configured users.
-Currently CDAP supports configuring impersonation at a namespace level, which means that every namespace will have
-a single user that programs in that namespace will run as, and that resources will be accessed as.
+Impersonation allows users to run programs and access datasets, streams, and other
+resources as pre-configured users. Currently CDAP supports configuring impersonation
+at a namespace level, which means that every namespace will have a single user that
+programs in that namespace will run as, and that resources will be accessed as.
+
+Requirements
+============
+To utilize this feature, Kerberos must be enabled on the cluster and configured in
+:ref:`cdap-site.xml <appendix-cdap-site.xml>`, via the parameter ``kerberos.auth.enabled``.
+To configure a namespace to have impersonation, specify the ``principal`` and
+``keytabURI`` in the :ref:`namespace configuration <http-restful-api-namespace-configs>`.
+
+If these are not specified, the principal and keytab of the CDAP Master user will be used.
+
+The configured principal must have been granted permissions for the operations that will
+occur in that namespace. For instance, if
+a :ref:`custom HBase namespace <namespaces-custom-mapping>` is configured, the configured
+principal must have privileges to create tables within that HBase namespace. If no
+custom HBase namespace is specified, the configured principal must have privilege to
+create namespaces.
+Because of this, it is simplest to specify a custom mapping for ``root.directory`` and
```

Comment:
It is not strictly required to specify custom mapping, but its much simpler to do so.